modules/core/src/main/java/org/apache/ignite/internal/jdbc2/JdbcStatement.java

```diff
@@ -142,7 +143,7 @@ else if (!conn.isMultipleStatementsAllowed() && conn.isMultipleStatementsTaskV2S
             }
         }
         catch (Exception e) {
-            throw convertToSqlException(e, "Failed to query Ignite.");
+            throw convertToSqlException(e, e instanceof SecurityException ? e.getMessage() : "Failed to query Ignite.");
```

Comment:
Can we move "e instanceof SecurityException ? e.getMessage()" into convertToSqlException()  ?